### PR TITLE
[ISSUE #1689]add interfaces to remove unused statsItems in BrokerStatsManager

### PR DESCRIPTION
**Pull request by zhanghaou**
## What is the purpose of the change

Fix ISSUE 1689, remove unused statsItems when topic or group deleted.

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: common/src/main/java/org/apache/rocketmq/common/stats/StatsItemSet.java [hunk 1]
```diff
@@ -158,6 +158,43 @@ public void addValue(final String statsKey, final int incValue, final int incTim
...
     }
 
+    public void delValue(final String statsKey) {
+        StatsItem statsItem = this.statsItemTable.get(statsKey);
+        if (null != statsItem) {
+            this.statsItemTable.remove(statsKey);
+        }
+    }
+
+    public void delValueByFirstKey(final String statsKey, String separator) {
```
Comment (zongtanghu) [inline common/src/main/java/org/apache/rocketmq/common/stats/StatsItemSet.java]: I thinks the method name "delValueByFirstKey" may not be suitable.And may be you can rename it, such as "delValueByPrefixKey";

File: common/src/main/java/org/apache/rocketmq/common/stats/StatsItemSet.java [hunk 2]
```diff
@@ -158,6 +158,43 @@ public void addValue(final String statsKey, final int incValue, final int incTim
...
+        Iterator<Entry<String, StatsItem>> it = this.statsItemTable.entrySet().iterator();
+        while (it.hasNext()) {
+            Entry<String, StatsItem> next = it.next();
+            if (next.getKey().contains(separator + statsKey + separator)) {
+                it.remove();
+            }
+        }
+    }
+
+    public void delValueByLastKey(final String statsKey, String separator) {
```
Comment (zongtanghu) [inline common/src/main/java/org/apache/rocketmq/common/stats/StatsItemSet.java]: the same as above.

File: common/src/main/java/org/apache/rocketmq/common/stats/StatsItemSet.java [hunk 3]
```diff
@@ -158,6 +158,43 @@ public void addValue(final String statsKey, final int incValue, final int incTim
...
+        Iterator<Entry<String, StatsItem>> it = this.statsItemTable.entrySet().iterator();
+        while (it.hasNext()) {
+            Entry<String, StatsItem> next = it.next();
+            if (next.getKey().startsWith(statsKey + separator)) {
+                it.remove();
+            }
+        }
+    }
+
+    public void delValueByMidKey(final String statsKey, String separator) {
```
Comment (zongtanghu) [inline common/src/main/java/org/apache/rocketmq/common/stats/StatsItemSet.java]: the same as above

**Comment by zongtanghu** [comment]
@duhenglucky @RongtongJin please help to review this pr together.

File: common/src/main/java/org/apache/rocketmq/common/stats/StatsItemSet.java (hunk 1 above)
Comment (zhanghaou) [reply common/src/main/java/org/apache/rocketmq/common/stats/StatsItemSet.java]: I have renamed all.

**Comment by zongtanghu** [comment]
It looks good, I approve this pr.

**Comment by zhanghaou** [comment]
> It would be better to provide an option for compatibility.

Does it mean that we need to add a switch when call these interfaces?

**Comment by zhanghaou** [comment]
@duhenglucky Is something wrong with coverage test?

**Comment by duhenglucky** [comment]
> @duhenglucky Is something wrong with coverage test?

nothing except the unit test coverage.

**Pull request closed without merging**